### ISSUE TITLE
(ctc_loss_calculator.cc) & (Infinite loss) & (Loss is not finite)

### ISSUE THREAD

**Issue by mrocr** [Calamari-OCR/calamari#89]
@ChWick Thank you for your hard work
While running the training command below, I keep getting errors constantly.
- Is this normal? 
- Are the lines too long?
- Does this mean that ctc_loss can't can't find a mapping between text labels and input?
- Will this slow down the training process?

```
(calamari) C:\Users\Home\program\calamari-master>calamari-train --files ./calamari_ocr/test/data/uw3_50lines/train/*.png --batch_size=4 --display=200 --output_dir ./store --checkpoint_frequency 1000 --validation ./calamari_ocr/test/data/uw3_50lines/test/*.png --early_stopping_frequency=200 --early_stopping_nbest=30
```

The errors:
```
tensorflow/core/util/ctc/ctc_loss_calculator.cc:144] No valid path found.
WARNING: Infinite loss. Skipping batch.
```
```
Error: Loss is not finite! Trying to restart from last checkpoint.WARNING:tensorflow:From C:\Users\Home\.conda\envs\calamari\lib\site-packages\tensorflow\python\training\saver.py:1266: checkpoint_exists (from tensorflow.python
.training.checkpoint_management) is deprecated and will be removed in a future version.
```

```
Windows 7 sp1 64bit
Calamari installed using environment_master_gpu.yml
```

**Comment by ChWick** [comment]
This should occur only very very rarely during the training process and sometimes requires to start the training anew.

Answers to your questions:
* No
* Usually not
* Yes
* Yes (might never learn anything)

Possible reasons:
* Mismatches between line image and text (errors in the GT)
* Destroyed input data
* Too high weight updates, e.g. cause by a learning rate that is too high

Suggestions:
* Check your data
* Reduce the learning rate (`--learning_rate`) e.g. by a factor of 10

At what iteration does the infinite loss occur? At the beginning, or during the training, i.e. the network already learned 'something'
